model/task_queue.go

```diff
@@ -32,18 +32,29 @@ type TaskGroupInfo struct {
 }
 
 type DistroQueueInfo struct {
-	Length             int             `bson:"length" json:"length"`
-	ExpectedDuration   time.Duration   `bson:"expected_duration" json:"expected_duration"`
-	CountOverThreshold int             `bson:"count_over_threshold" json:"count_over_threshold"`
-	TaskGroupInfos     []TaskGroupInfo `bson:"task_group_infos" json:"task_group_infos"`
+	Length               int             `bson:"length" json:"length"`
+	ExpectedDuration     time.Duration   `bson:"expected_duration" json:"expected_duration"`
+	MaxDurationThreshold time.Duration   `bson:"max_duration_threshold" json:"max_duration_threshold"`
+	CountOverThreshold   int             `bson:"count_over_threshold" json:"count_over_threshold"`
+	TaskGroupInfos       []TaskGroupInfo `bson:"task_group_infos" json:"task_group_infos"`
+}
+
+func GetDistroQueueInfo(distroID string) (DistroQueueInfo, error) {
+	taskQueue := &TaskQueue{}
+	query := bson.M{taskQueueDistroKey: distroID}
+	projection := bson.M{taskQueueDistroQueueInfoKey: 1, taskQueueIdKey: 0}
+	err := db.FindOne(TaskQueuesCollection, query, projection, db.NoSort, taskQueue)
```

Comment:
nit: combine lines 44-46 and wrap the error.

Reply:
Done